### ISSUE TITLE
Data Store What is `Unsupported element '$[filter][and]'.` error?

### ISSUE THREAD

**Issue by matart15** [aws-amplify/amplify-js#9623]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

DataStore

### Environment information

<details>

```
# Put output below this line


  System:
    OS: macOS 12.2.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 30.21 GB / 64.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.22.1 - ~/.nvm/versions/node/v12.22.1/bin/node
    Yarn: 1.22.17 - ~/.yvm/shim/yarn
    npm: 8.3.0 - ~/.nvm/versions/node/v12.22.1/bin/npm
    Watchman: 2021.10.18.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 98.0.4758.102
    Firefox: 97.0
    Safari: 15.3
  npmGlobalPackages:
    @aws-amplify/cli: 7.6.21
    @nestjs/cli: 8.1.5
    @teambit/bvm: 0.0.34
    aws-cdk: 2.3.0
    detox-cli: 19.0.0
    eas-cli: 0.47.0
    expo-cli: 5.2.0
    http-server: 14.1.0
    npm: 8.3.0
    nx: 13.2.3
    pm2: 5.1.2
    prisma: 3.5.0


```

</details>

### Describe the bug

The app and backend were working fine a few hours ago. 

Suddenly this error. No backend change. ( also my git log says no change )

`Unsupported element '$[filter][and]'.`

What is this error? A quick google says it's some kind of DynamoDB related. but not sure where to look

### Expected behavior

just work

### Reproduction steps

I have no Idea

### Additional information and screenshots

schema.graphql

```graphql
enum ProjectStatus {
  OPEN
  ONGOING
  ARCHIVED
  CLOSED
}

type Project1 @model @auth(rules: [{allow: groups, groups: ["Admin"]}, {allow: groups, groupsField: "company", operations: [read]}]) {
  id: ID!
  name: String
  status: ProjectStatus
  company: String
}

type Door @model @auth(rules: [{allow: public}, {allow: private}]) {
  id: ID!
  name: String!
  allowedCompanies: [String]
  daitimeUnlockAllowedMembers: [String]
  httpEndpoint: String!
  serviceUuid: String!
  index: Int!
  imageUrl: String!
  isOpenDaily: Boolean!
  isOpenNow: Boolean
}

```

**Comment by chrisbonifacio** [comment]
Hi @matart15 👋 To get some more context around this issue, can you share some code snippets of how you might have configured DataStore in your application (ex. using any sync expressions, etc) as well as how you are saving and querying data?

What code is being run that leads to this error? Does it happen on save or query, or does it just appear in the console (perhaps during a sync)? 

You can also enable logging in your app for Amplify by adding this line above or below your `Amplify.configure`

```js
Amplify.Logger.LOG_LEVEL = 'DEBUG';
```

Lastly, can you share the contents of your package.json file to see what packages and versions of Amplify you're using?

**Comment by matart15** [comment]
I debugged and found that `Datastore.start()` is causing the issue.

Even I tried deleting the whole app and reinstalling, clearing js cache, yarn cache. all results same.
Also tried with and withoud SQLiteAdapter

my configuration Code.
<details>

```
window.LOG_LEVEL = 'DEBUG'
// Apply plugin with configuration
Amplify.addPluggable(
  new AWSIoTProvider({
    aws_pubsub_region: 'ap-northeast-1',
    aws_pubsub_endpoint:
      'wss://aunlh4rgvg4aq-ats.iot.ap-northeast-1.amazonaws.com/mqtt',
    clientId: 'goldware_raspi',
  }),
)

// DataStore.configure({
//   syncExpressions: [
//     // @ts-ignore
//     syncExpression(Project1, (): ((p: Project1) => unknown) => {
//       return (p: Project1) => !p
//     }),
//     // @ts-ignore
//     syncExpression(Door, (): ((post: Door) => unknown) => {
//       return (post: Door) => post
//     }),
//   ],
// })
Amplify.configure(awsconfig)
// DataStore.configure({
//   storageAdapter: SQLiteAdapter,
// })
```

</details>


Then I found out that creating a new expo app and copying all ts files including App.ts, tsconfig worked.

the only difference is the `@expo/*` packages.

both project using same amplify aws-exports file


- not working ( react native project )

<details>

```
  System:
    OS: macOS 12.2.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 30.80 GB / 64.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.22.1 - ~/.nvm/versions/node/v12.22.1/bin/node
    Yarn: 1.22.17 - ~/.yvm/shim/yarn
    npm: 8.3.0 - ~/.nvm/versions/node/v12.22.1/bin/npm
    Watchman: 2021.10.18.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 98.0.4758.109
    Firefox: 97.0.1
    Safari: 15.3
  npmPackages:
    @babel/core: 7.17.5 => 7.17.5 
    @babel/plugin-transform-arrow-functions: 7.16.7 => 7.16.7 
    @babel/plugin-transform-shorthand-properties: 7.16.7 => 7.16.7 
    @babel/runtime: 7.12.5 => 7.12.5 (7.17.2)
    @eva-design/eva: 2.1.1 => 2.1.1 
    @react-native-async-storage/async-storage: 1.16.1 => 1.16.1 
    @react-native-community/eslint-config: 2.0.0 => 2.0.0 
    @react-native-community/netinfo: 8.0.0 => 8.0.0 
    @react-native-picker/picker: 2.3.0 => 2.3.0 
    @react-navigation/drawer: 6.3.1 => 6.3.1 
    @react-navigation/native: 6.0.8 => 6.0.8 
    @react-navigation/native-stack: 6.5.0 => 6.5.0 
    @types/jest: 26.0.23 => 26.0.23 
    @types/react-native: 0.66.15 => 0.66.15 (0.66.16)
    @types/react-native-snap-carousel: 3.8.5 => 3.8.5 
    @types/react-test-renderer: 17.0.1 => 17.0.1 
    @typescript-eslint/eslint-plugin: 5.7.0 => 5.7.0 (3.10.1)
    @typescript-eslint/parser: 5.7.0 => 5.7.0 (3.10.1)
    @ui-kitten/components: 5.1.1 => 5.1.1 
    @ui-kitten/eva-icons: 5.1.1 => 5.1.1 
    HelloWorld:  0.0.1 
    amazon-cognito-identity-js: 5.2.6 => 5.2.6 
    aws-amplify: 4.3.14 => 4.3.14 
    aws-amplify-react-native: 6.0.2 => 6.0.2 
    babel-jest: 26.6.3 => 26.6.3 
    eslint: 8.9.0 => 8.9.0 
    eslint-config-airbnb: 19.0.4 => 19.0.4 
    eslint-config-prettier: 8.3.0 => 8.3.0 (6.15.0)
    eslint-import-resolver-typescript: 2.5.0 => 2.5.0 
    eslint-plugin-autofix: 1.1.0 => 1.1.0 
    eslint-plugin-flowtype: 8.0.3 => 8.0.3 (2.50.3)
    eslint-plugin-graphql: 4.0.0 => 4.0.0 
    eslint-plugin-import: 2.25.4 => 2.25.4 
    eslint-plugin-prettier: 4.0.0 => 4.0.0 (3.1.2)
    eslint-plugin-react: 7.28.0 => 7.28.0 
    eslint-plugin-react-native: 4.0.0 => 4.0.0 (3.11.0)
    eslint-plugin-simple-import-sort: 7.0.0 => 7.0.0 
    eslint-plugin-unused-imports: 2.0.0 => 2.0.0 
    hermes-inspector-msggen:  1.0.0 
    husky: 7.0.4 => 7.0.4 
    jest: 26.6.3 => 26.6.3 
    lint-staged: 12.3.4 => 12.3.4 
    metro-react-native-babel-preset: 0.66.2 => 0.66.2 
    prettier: 2.5.1 => 2.5.1 
    react: 17.0.2 => 17.0.2 
    react-native: 0.67.2 => 0.67.2 
    react-native-fast-image: 8.5.11 => 8.5.11 
    react-native-gesture-handler: 2.2.0 => 2.2.0 
    react-native-keyboard-aware-scroll-view: 0.9.5 => 0.9.5 
    react-native-linear-gradient: 2.5.6 => 2.5.6 
    react-native-reanimated: 2.4.1 => 2.4.1 
    react-native-safe-area-context: 3.4.1 => 3.4.1 
    react-native-screens: 3.12.0 => 3.12.0 
    react-native-size-matters: 0.4.0 => 0.4.0 
    react-native-snap-carousel: 3.9.1 => 3.9.1 
    react-native-svg: 12.1.1 => 12.1.1 
    react-test-renderer: 17.0.2 => 17.0.2 
    typescript: 4.5.5 => 4.5.5 
  npmGlobalPackages:
    @aws-amplify/cli: 7.6.21
    @nestjs/cli: 8.1.5
    @react-native-community/cli: 7.0.3
    @teambit/bvm: 0.0.34
    aws-cdk: 2.3.0
    detox-cli: 19.0.0
    eas-cli: 0.47.0
    expo-cli: 5.2.0
    http-server: 14.1.0
    npm: 8.3.0
    nx: 13.2.3
    pm2: 5.1.2
    prisma: 3.5.0


```

</details>

- working ( expo project. bare workflow )

<details>

```
  System:
    OS: macOS 12.2.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 30.74 GB / 64.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 12.22.1 - ~/.nvm/versions/node/v12.22.1/bin/node
    Yarn: 1.22.17 - ~/.yvm/shim/yarn
    npm: 8.3.0 - ~/.nvm/versions/node/v12.22.1/bin/npm
    Watchman: 2021.10.18.00 - /usr/local/bin/watchman
  Browsers:
    Chrome: 98.0.4758.109
    Firefox: 97.0.1
    Safari: 15.3
  npmPackages:
    @babel/core: 7.17.4 => 7.17.4 (7.17.5)
    @eva-design/eva: 2.1.1 => 2.1.1 
    @expo/vector-icons: 12.0.0 => 12.0.0 (12.0.5)
    @react-native-async-storage/async-storage: 1.16.1 => 1.16.1 
    @react-native-community/eslint-config: 3.0.1 => 3.0.1 
    @react-native-community/netinfo: 7.1.3 => 7.1.3 
    @react-navigation/drawer: 6.3.1 => 6.3.1 
    @react-navigation/native: 6.0.2 => 6.0.2 
    @react-navigation/native-stack: 6.1.0 => 6.1.0 
    @types/jest: 27.4.0 => 27.4.0 
    @types/react: 17.0.39 => 17.0.39 
    @types/react-native: 0.64.12 => 0.64.12 (0.66.16)
    @types/react-native-snap-carousel: 3.8.5 => 3.8.5 
    @typescript-eslint/eslint-plugin: 5.12.0 => 5.12.0 (4.33.0)
    @typescript-eslint/eslint-plugin-tslint: 5.12.0 => 5.12.0 
    @typescript-eslint/parser: 5.12.0 => 5.12.0 (4.33.0)
    @ui-kitten/components: 5.1.1 => 5.1.1 
    @ui-kitten/eva-icons: 5.1.1 => 5.1.1 
    HelloWorld:  0.0.1 
    amazon-cognito-identity-js: 5.2.6 => 5.2.6 
    aws-amplify: 4.3.14 => 4.3.14 
    aws-amplify-react-native: 6.0.2 => 6.0.2 
    babel-jest: 27.5.1 => 27.5.1 (26.6.3)
    eslint: 8.9.0 => 8.9.0 
    eslint-config-airbnb: 19.0.4 => 19.0.4 
    eslint-config-prettier: 8.3.0 => 8.3.0 (6.15.0)
    eslint-import-resolver-typescript: 2.5.0 => 2.5.0 
    eslint-plugin-autofix: 1.1.0 => 1.1.0 
    eslint-plugin-flowtype: 8.0.3 => 8.0.3 (2.50.3)
    eslint-plugin-graphql: 4.0.0 => 4.0.0 
    eslint-plugin-import: 2.25.4 => 2.25.4 
    eslint-plugin-prettier: 4.0.0 => 4.0.0 (3.1.2)
    eslint-plugin-react: 7.28.0 => 7.28.0 
    eslint-plugin-react-native: 4.0.0 => 4.0.0 (3.11.0)
    eslint-plugin-simple-import-sort: 7.0.0 => 7.0.0 
    eslint-plugin-unused-imports: 2.0.0 => 2.0.0 
    expo: 44.0.0 => 44.0.0 
    expo-asset: 8.4.6 => 8.4.6 
    expo-constants: 13.0.1 => 13.0.1 (13.0.2)
    expo-dev-client: ~0.8.4 => 0.8.4 
    expo-font: 10.0.4 => 10.0.4 (10.0.5)
    expo-linking: 3.0.0 => 3.0.0 
    expo-secure-store: ~11.1.0 => 11.1.1 
    expo-splash-screen: 0.14.1 => 0.14.1 
    expo-status-bar: 1.2.0 => 1.2.0 
    expo-updates: ~0.11.6 => 0.11.6 
    expo-web-browser: 10.1.0 => 10.1.0 
    formik: 2.2.9 => 2.2.9 
    hermes-inspector-msggen:  1.0.0 
    husky: 7.0.4 => 7.0.4 
    jest: 26.6.3 => 26.6.3 
    jest-expo: 44.0.1 => 44.0.1 
    lint-staged: 12.3.4 => 12.3.4 
    react: 17.0.1 => 17.0.1 
    react-dom: 17.0.1 => 17.0.1 
    react-native: 0.64.3 => 0.64.3 
    react-native-fast-image: 8.5.11 => 8.5.11 
    react-native-gesture-handler: ~2.1.0 => 2.1.3 
    react-native-keyboard-aware-scroll-view: 0.9.5 => 0.9.5 
    react-native-linear-gradient: 2.5.6 => 2.5.6 
    react-native-reanimated: ~2.3.1 => 2.3.2 
    react-native-safe-area-context: 3.3.2 => 3.3.2 
    react-native-screens: 3.10.1 => 3.10.1 (3.12.0)
    react-native-size-matters: 0.4.0 => 0.4.0 
    react-native-snap-carousel: 3.9.1 => 3.9.1 
    react-native-svg: 12.1.1 => 12.1.1 
    react-native-web: 0.17.1 => 0.17.1 
    react-test-renderer: 17.0.2 => 17.0.2 
    typescript: 4.5.5 => 4.5.5 
  npmGlobalPackages:
    @aws-amplify/cli: 7.6.21
    @nestjs/cli: 8.1.5
    @react-native-community/cli: 7.0.3
    @teambit/bvm: 0.0.34
    aws-cdk: 2.3.0
    detox-cli: 19.0.0
    eas-cli: 0.47.0
    expo-cli: 5.2.0
    http-server: 14.1.0
    npm: 8.3.0
    nx: 13.2.3
    pm2: 5.1.2
    prisma: 3.5.0


```

</details>

**Comment by chrisbonifacio** [comment]
@matart15 If it seems to be working on a fresh app, there should be some difference between either the shape of the graphql query or the resolver.

Can you share what the graphql query looks like in the network request on the app that's giving you that error?
Also, the resolver template would be helpful, which you can find by navigating to this directory in your project: `amplify/backend/api/<apiName>/build/resolvers/Query.syncDoors.req.vtl`

**Comment by chrisbonifacio** [comment]
Hey @matart15 , just wanted to follow up with you and see if you're still experiencing this issue.

**Comment by matart15** [comment]
I gave up on amplify. stopped using it. you can close the issue.

**Comment by chrisbonifacio** [comment]
@matart15 Sorry to hear that. If you'd like to leave some feedback on what your pain points were with our libraries or the quality of support provided here, please feel free to do so. If there were any guides and/or pages of our documentation that you were following but felt difficult to understand, please share those as well.

I can share it with the team to improve our developer experience.

Otherwise, if you decide to give it another try in the future, we do have a pretty active community of developers on [Discord](https://discord.com/invite/amplify), consisting of all levels of experience.

**Comment by matart15** [comment]
@chrisbonifacio  Thank you. 

I left the amplify. but just in case writing some feedback

1. Datastore is unstable. ( @chrisbonifacio  and I had 3 active problems, 2 solved 1 remained. ) mostly does not work as docs said. At first, I thought of my implementation problem. But I spent 7 days on this. tried every possible way. So now I am thinking the library is just unstable.
2. Pubsub is unstable too. ( I created 2 issues. In a week not resolved any of it ). 
3. Datastore and Pubsub are the only services Amplify provides outside AWS services. others are just wrappers. ( maybe pubsub too ). If this 2 does not work well, its better just go with CDK or pulumi 
4. Requires `OpenSearch`. No pagination by default. There is a `@searchable`, which is connected with `OpenSearch`. The problem with OpenSearch is it duplicates data. It costs an extra fee. If I use OpenSearch I can't call any dynamodb api. which will result in a broken search query.
6. Infra errors are hard to debug. Especially errors have very little information.
7. Most frustrating part was when something happens I don't know it was my implementation fault or there are some open bugs about it. I have to search read a lot of issues.

Overall My assumption is most people would choose to `amplify` to reduce development time, but in reality, it increases it.
